internal/brokerstore/brokerstore.go

```diff
@@ -115,29 +117,46 @@ func (b *BrokerStore) GetConsumer(ctx context.Context, id string, op messagebrok
 	if perr != nil {
 		return nil, perr
 	}
+
+	brokerStoreActiveConsumersCount.WithLabelValues(env, key.String()).Inc()
+
 	consumer, _ = b.consumerMap.LoadOrStore(key.String(), newConsumer)
 	b.partitionLock.Unlock(key.String()) // unlock
 	return consumer.(messagebroker.Consumer), nil
 }
 
 // RemoveConsumer deletes the consumer from the store
-func (b *BrokerStore) RemoveConsumer(_ context.Context, id string, op messagebroker.ConsumerClientOptions) bool {
+func (b *BrokerStore) RemoveConsumer(ctx context.Context, id string, op messagebroker.ConsumerClientOptions) bool {
+	logger.Ctx(ctx).Infow("brokerstore: request to close consumer", "id", id, "groupID", op.GroupID)
+
 	brokerStoreOperationCount.WithLabelValues(env, "RemoveConsumer").Inc()
 
 	startTime := time.Now()
-	defer brokerStoreOperationTimeTaken.WithLabelValues(env, "RemoveConsumer").Observe(float64(time.Since(startTime).Nanoseconds() / 1e9))
+	defer brokerStoreOperationTimeTaken.WithLabelValues(env, "RemoveConsumer").Observe(time.Now().Sub(startTime).Seconds())
+
+	wasConsumerFound := false
 
 	key := NewKey(op.GroupID, id)
-	_, loaded := b.consumerMap.LoadAndDelete(key)
-	return loaded
+	b.partitionLock.Lock(key.String())         // lock
+	defer b.partitionLock.Unlock(key.String()) // unlock
+
+	consumer, ok := b.consumerMap.Load(key.String())
+	if ok {
+		wasConsumerFound = true
+		b.consumerMap.Delete(consumer)
+		brokerStoreActiveConsumersCount.WithLabelValues(env, key.String()).Dec()
+	}
+
+	logger.Ctx(ctx).Infow("brokerstore: consumer removal completed", "id", id, "group_id", op.GroupID, "was_consumer_found", wasConsumerFound)
```

Comment:
this log can be confusing, should we log 2 separately

1. consumer not found
2. consumer removed successfully

Reply:
done